### PR TITLE
Add Chart Filter Loading Indicator

### PR DESCRIPTION
**Pull request by iAmKevinMcKee**
I added the loading indicator as discussed in Discord. A few things to note:

I added a public property and set it to `true`. Not sure if you want the default true or false. Also, I'm not sure if you want the long name I gave it. Of course you are free to edit whatever.

I played around with adding a `wire:loading.delay` but even with `wire:loading.delay.shortest` it felt janky. I think it is best as is and if someone doesn't want it, they can turn it off. Again, always up to you.

Finally, not sure if you are good with the svg being inline or if you want that loading indicator extracted as a component. If you want to do this, please let me know where you want that component to be and I can update.

If this is merged I'm happy to update docs, but wasn't sure what kind of changes you'd want to make before doing that.

### CODE REVIEW

**Comment by zepfietje** [comment]
@iAmKevinMcKee, there already exists a loading indicator component being used in a lot of places (`<x-filament::loading-indicator>`.
Just search for it to find usages.
Could you update this PR to use that component? 😄

**Comment by iAmKevinMcKee** [comment]
Got it. Made that update. Thanks for the feedback.

File: packages/widgets/resources/views/chart-widget.blade.php [hunk 1]
```diff
@@ -23,17 +23,27 @@
...
-                        wire:model="filter"
-                        wire:loading.class="animate-pulse"
-                    >
-                        @foreach ($filters as $value => $label)
-                            <option value="{{ $value }}">
-                                {{ $label }}
-                            </option>
-                        @endforeach
-                    </select>
+                    <div class="flex items-center space-x-3">
```
Comment (zepfietje) [inline packages/widgets/resources/views/chart-widget.blade.php]:
```suggestion
                    <div class="flex items-center gap-3">
```

File: packages/widgets/resources/views/chart-widget.blade.php [hunk 2]
```diff
@@ -23,17 +23,27 @@
...
+                            wire:model="filter"
+                            wire:loading.class="animate-pulse"
+                        >
+                            @foreach ($filters as $value => $label)
+                                <option value="{{ $value }}">
+                                    {{ $label }}
+                                </option>
+                            @endforeach
+                        </select>
+                    </div>
```
Comment (zepfietje) [inline packages/widgets/resources/views/chart-widget.blade.php]:
```suggestion
                        @if ($showFilterLoadingIndicator)
                            <x-filament::loading-indicator
                                x-cloak="x-cloak"
                                class="h-8 w-8 text-gray-500"
                                wire:loading
                                wire:target="filter"
                            />
                        @endif

                        <select
                            class="text-gray-900 border-gray-300 block h-10 transition duration-75 rounded-lg shadow-sm focus:border-primary-500 focus:ring-1 focus:ring-inset focus:ring-primary-500 dark:bg-gray-700 dark:border-gray-600 dark:text-gray-200 dark:focus:border-primary-500"
                            wire:model="filter"
                            wire:loading.class="animate-pulse"
                        >
                            @foreach ($filters as $value => $label)
                                <option value="{{ $value }}">
                                    {{ $label }}
                                </option>
                            @endforeach
                        </select>
                    </div>
```

**Review by zepfietje — APPROVED** [review]

File: packages/widgets/resources/views/chart-widget.blade.php [hunk 3]
```diff
@@ -23,17 +23,28 @@
...
-                            <option value="{{ $value }}">
-                                {{ $label }}
-                            </option>
-                        @endforeach
-                    </select>
+                    <div class="flex items-center gap-3">
+                        @if ($showFilterLoadingIndicator)
+                            <x-filament::loading-indicator
+                                x-cloak="x-cloak"
+                                class="h-8 w-8 text-gray-500"
```
Comment (zepfietje) [inline packages/widgets/resources/views/chart-widget.blade.php:30]: These size values look odd to me? (Haven't tested visually, but just from the code.)

**Review by zepfietje — CHANGES_REQUESTED** [review]

File: packages/widgets/resources/views/chart-widget.blade.php (hunk 3 above)
Comment (zepfietje) [reply packages/widgets/resources/views/chart-widget.blade.php:30]: Also not sure about the color.

Comment (iAmKevinMcKee) [reply packages/widgets/resources/views/chart-widget.blade.php:30]: Here's what it looks like: 
image

File: packages/widgets/src/ChartWidget.php [hunk 4]
```diff
@@ -14,6 +14,8 @@ abstract class ChartWidget extends Widget
     protected ?array $cachedData = null;
 
     public string $dataChecksum;
+    
+    public bool $showFilterLoadingIndicator = true;
```
Comment (danharrin) [inline packages/widgets/src/ChartWidget.php]:
```suggestion
    public bool $hasFilterLoadingIndicator = true;
```

File: packages/widgets/resources/views/chart-widget.blade.php [hunk 5]
```diff
@@ -23,17 +23,28 @@
...
-                        wire:loading.class="animate-pulse"
-                    >
-                        @foreach ($filters as $value => $label)
-                            <option value="{{ $value }}">
-                                {{ $label }}
-                            </option>
-                        @endforeach
-                    </select>
+                    <div class="flex items-center gap-3">
+                        @if ($showFilterLoadingIndicator)
```
Comment (danharrin) [inline packages/widgets/resources/views/chart-widget.blade.php]:
```suggestion
                        @if ($hasFilterLoadingIndicator)
```

**Pull request merged**